app/customer.py

```diff
@@ -0,0 +1,64 @@
+import dataclasses
+import math
+
+from app.shop import Shop
+
+
+@dataclasses.dataclass
+class Customer:
+    name: str
+    location: list[int]
+    money: int
+    car: dict
+    product_cart: dict
+
+    @staticmethod
+    def fuel_to_shop(
+            fuel_cons: float, distance_to_shop: float, fuel_price: float
+    ) -> float:
+        return ((fuel_cons * distance_to_shop) / 100 * fuel_price) * 2
+
+    def dist_to_shop(self, shop_location: list) -> float:
+        return (
+            math.sqrt((shop_location[0] - self.location[0]) ** 2
+                      + (shop_location[1] - self.location[1]) ** 2)
+        )
+
+    def is_all_products_exist(self) -> bool:
+        return all(
+            product in Shop.products.keys()
+            for product in self.product_cart.keys()
+        )
+
+    def product_total(self, products: dict) -> float:
+        if self.is_all_products_exist:
+            return sum(
+                count * products[name]
+                for name, count in self.product_cart.items()
+            )
+
+    def fuel_expenses(
+            self, fuel: float, shop_location: list[int], fuel_price: float
+    ) -> float:
+        fuel_total = round(
+            self.fuel_to_shop(
+                fuel, self.dist_to_shop(shop_location), fuel_price
+            ), 2
+        )
+        return fuel_total
+
+    def shop_choice(
+            self, shop_names: dict, closest_shop: str
+    ) -> float:
+        total_costs = 0
+        for product_name, product_count in self.product_cart.items():
+            if product_name in shop_names[closest_shop]:
```

Comment:
```suggestion
            if product_name == shop_names[closest_shop]:
```

Reply:
in such case output will be as follows:

<img width="326" alt="Screenshot 2023-03-06 at 17 55 10" src="https://user-images.githubusercontent.com/10571459/223162374-d8ef24da-b1a9-405a-99c5-3a49587ec139.png">